libvast/src/query_queue.cpp

```diff
@@ -62,18 +77,17 @@ query_queue::insert(query_state&& query_state, std::vector<uuid>&& candidates) {
     return caf::make_error(ec::unspecified, "A query with this ID exists "
                                             "already");
   for (const auto& cand : candidates) {
-    auto it = std::find_if(partitions.begin(), partitions.end(), [&](auto& x) {
-      return x.partition == cand;
-    });
+    auto it = std::find(partitions.begin(), partitions.end(), cand);
     if (it != partitions.end()) {
       it->priority += query_state_it->second.query.priority;
       it->queries.push_back(qid);
+      VAST_ASSERT_CHEAP(!detail::contains(inactive_partitions, cand),
+                        "A partition must not be active and inactive at the "
+                        "same time");
```

Comment:
Isn't this check being executed `candidates * inactive-partitions` times? I wouldn't call this cheap.